source/mixins/versionable.js

```diff
@@ -0,0 +1,19 @@
+Space.messaging.Versionable = {
+
+  schemaVersion: 1,
+
+  onConstruction(data) {
+    if (_.isObject(data)) this._migrateToLatestVersion(data);
+  },
+
+  _migrateToLatestVersion(data) {
+    let schemaVersion = data.schemaVersion;
+    if (!schemaVersion || schemaVersion === this.schemaVersion) return;
+    for (let version = schemaVersion; version < this.schemaVersion; version++) {
+      let migrateMethod = this[`migrateFromVersion${version}`];
+      if (migrateMethod !== undefined) migrateMethod.call(this, data);
```

Comment:
Maybe we should throw an error if migration method is undefined and force having migration methods for every version?